src/App.tsx

```diff
@@ -1,25 +1,33 @@
-import React from 'react';
-import logo from './logo.svg';
-import './App.css';
+import "./App.css";
+import Counter from "./components/Counter";
+import Search from "./components/Search";
+import Genre from "./components/Genre";
 
 function App() {
+  function onSearch(searchText: string) {
+    console.log(searchText);
+  }
+
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.tsx</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
+    <>
+      <h1>Counters</h1>
+      <Counter initialValue={0}></Counter>
+      <Counter initialValue={10}></Counter>
+      <h1>SearchForm</h1>
+      <Search intitialSearchText="Search" onSearch={onSearch}></Search>
+      <Search intitialSearchText="test" onSearch={onSearch}></Search>
+      <h1>GenreSelect</h1>
+      <Genre
+        genres={["Action", "Comedy", "Romance"]}
+        selectedGenre="Comedy"
+        onSelect={(genre) => console.log(genre)}
```

Comment:
Try to avoid as much as you can usage of inline anonymous functions in JSX, particularly when passing them as props. With each component rerender, a new function is created, potentially causing unnecessary rerenders for children components that receive this function as a prop.

Reply:
Changed to function